src/libDirectoryService/ViewChangePostProcessing.cpp

```diff
@@ -173,16 +172,15 @@ void DirectoryService::ProcessViewChangeConsensusWhenDone()
     unsigned int offsetToCandidateLeader = 1;
 
     Peer expectedLeader;
-    if (m_mediator.m_DSCommitteeNetworkInfo.at(offsetToCandidateLeader)
-        == Peer())
+    if (m_mediator.m_DSCommittee.at(offsetToCandidateLeader).second == Peer())
```

Comment:
while this does not relate directly to this pr,  maybe we can add an extra check before this. Check whether offsetToCandidateLeader < m_DSCommittee,size() to prevent out of bound issue. This will become an issue once we randomly pick a leader.

Reply:
Since we haven't yet determined what the DS should do if the offset is somehow beyond the committee size, I propose we don't add a check yet if we can't add any corrective action along with it.

In any case, I have added a note about this in the random selection issue (https://github.com/Zilliqa/Issues/issues/103) so we don't forget this comment.